ignite/pkg/relayer/relayer.go

```diff
@@ -36,87 +36,111 @@ func New(ca cosmosaccount.Registry) Relayer {
 	}
 }
 
-// Link links all chains that has a path to each other.
+// LinkPaths links all chains that has a path from config file to each other.
 // paths are optional and acts as a filter to only link some chains.
 // calling Link multiple times for the same paths does not have any side effects.
-func (r Relayer) Link(ctx context.Context, pathIDs ...string) error {
+func (r Relayer) LinkPaths(
+	ctx context.Context,
+	pathIDs ...string,
+) error {
 	conf, err := relayerconf.Get()
 	if err != nil {
 		return err
 	}
 
 	for _, id := range pathIDs {
-		path, err := conf.PathByID(id)
+		conf, err = r.Link(ctx, conf, id)
 		if err != nil {
 			return err
 		}
-
-		if path.Src.ChannelID != "" { // already linked.
-			continue
-		}
-
-		if path, err = r.call(ctx, conf, path, "link"); err != nil {
-			return err
-		}
-
-		if err := conf.UpdatePath(path); err != nil {
-			return err
-		}
 		if err := relayerconf.Save(conf); err != nil {
 			return err
 		}
 	}
-
 	return nil
 }
 
-// Start relays packets for linked paths until ctx is canceled.
-func (r Relayer) Start(ctx context.Context, pathIDs ...string) error {
+// Link links chain path to each other.
+func (r Relayer) Link(
+	ctx context.Context,
+	conf relayerconf.Config,
+	pathID string,
+) (relayerconf.Config, error) {
+	path, err := conf.PathByID(pathID)
+	if err != nil {
+		return conf, err
+	}
+
+	if path.Src.ChannelID != "" {
+		return conf, fmt.Errorf("path %s already linked", path.ID)
+	}
+
+	if path, err = r.call(ctx, conf, path, "link"); err != nil {
+		return conf, err
+	}
+
+	return conf, conf.UpdatePath(path)
+}
+
+// StartPaths relays packets for linked paths from config file until ctx is canceled.
+func (r Relayer) StartPaths(ctx context.Context, pathIDs ...string) error {
 	conf, err := relayerconf.Get()
 	if err != nil {
 		return err
 	}
 
-	wg, ctx := errgroup.WithContext(ctx)
 	var m sync.Mutex // protects relayerconf.Path.
-
-	start := func(id string) error {
-		path, err := conf.PathByID(id)
+	for _, id := range pathIDs {
+		id := id
+		err := r.Start(ctx, conf, id, func(path relayerconf.Config) error {
+			m.Lock()
+			defer m.Unlock()
+			return relayerconf.Save(conf)
+		})
 		if err != nil {
 			return err
 		}
+	}
+	return nil
+}
 
-		if path, err = r.call(ctx, conf, path, "start"); err != nil {
+// Start relays packets for linked path until ctx is canceled.
+func (r Relayer) Start(
+	ctx context.Context,
+	conf relayerconf.Config,
+	pathID string,
+	postExecute func(path relayerconf.Config) error,
+) error {
+	wg, ctx := errgroup.WithContext(ctx)
+	start := func(id string) error {
```

Comment:
Seems like `id` is no longer used, once removed you can simplify the call to `DoNow` like that :

```go
	wg.Go(func() error {
		return ctxticker.DoNow(ctx, relayDuration, start)
	})
```

Reply:
[66bc68f](https://github.com/ignite/cli/pull/2516/commits/66bc68fb8c02240ec89cb16ee66679436a6a5f14)